tool/lib/commands/generate_changelog.dart

```diff
@@ -125,30 +125,34 @@ class GenerateChangelogCommand extends Command {
     print(
         'Adding entries to the changelog for all commits since $commitDate...');
 
-    // TODO(kenz): handle cases where there are more than 100 commits.
-    final uri = Uri.https(
-      '${auth}api.github.com',
-      '/repos/flutter/devtools/commits',
-      {
-        'since': commitDate,
-        'per_page': '100',
-      },
-    );
-    final commits = jsonDecode((await http.get(uri)).body);
+    const tryPages = 5;
     final changes = <String>[];
-    for (var commit in commits) {
-      if (commit['sha'] == taggedCommit['sha']) {
-        print(
-            'Skipping commit ${commit['sha']} because this is the commit of the previous tag');
-        continue;
-      }
-      final message = commit['commit']['message'];
-      if (_shouldSkip(commit['commit']['message'])) {
-        print('Skipping commit marked to be ignored: $message');
-        continue;
+    for (var i = 1; i <= tryPages; i++) {
```

Comment:
Used a while loop instead. Since this is just a convenience script, I added a TODO to consider traversing the response with pagination. If this while loop becomes buggy, we can spend time to do this the proper way.